eknigge/GuideWalker
# Added email/password authentication

# Description of Change

This PR addresses the following issues:

Closes #21  - Login Page

I pretty much copy/pasted the entire example code from here: [https://blog.jscrambler.com/how-to-integrate-firebase-authentication-with-an-expo-app](https://blog.jscrambler.com/how-to-integrate-firebase-authentication-with-an-expo-app)

I'm using @eknigge's **GuideWalkerTest** Firebase project for this, not Cheryl's.

Login page:

![Screenshot_1635723417](https://user-images.githubusercontent.com/62415298/139605082-29845697-b2de-4b3a-a5b5-ef2cf655b4fb.png)

------------
After login:

![Screenshot_1635723539](https://user-images.githubusercontent.com/62415298/139605089-b4e0a5ce-6fc3-4ada-952e-4b2e6b8cd494.png)

## How to test

 **⚠ You need to download the `.env` file I shared in Slack or you won't be connected to Firebase**

  1. Download the `.env` file I shared in Slack
  1. Put it inside the root folder of our repo
  1. Run `yarn install` to add new dependencies, there's a lot.
  1. `expo start` to run the app
  1. Run the app in Android emulator or iOS
  1. Sign up for an account
  1. Login with your account
  1. It should take you to the map page
  1. Clicking on the right right icon should sign you back out

## Checklist

- [x] Did you request review from someone?
